### PR TITLE
 added github based regression tests for glusterfs

### PR DESCRIPTION
**Pull request by amarts**
Signed-off-by: Amar Tumballi <amarts@gmail.com>

### CODE REVIEW

File: jobs/scripts/glusterfs/regression.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+echo "Start time $(date)"
+echo
+echo "Display all environment variables"
+echo "*********************************"
+echo
+echo "$MY_ENV"
+echo
+
+# Remove any gluster daemon leftovers from aborted runs
+sudo -E bash /opt/qa/cleanup.sh
```
Comment (deepshikhaaa) [inline jobs/scripts/glusterfs/regression.sh]: We don't have a /opt/qa setup on centos ci duffy nodes (which is dynamic allocation). You need to pull in the repo https://github.com/gluster/glusterfs-patch-acceptance-tests (similar to glusterfs) here in the node.

File: jobs/glusterfs-regression.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        name: ghprbPullId
+    - string:
+        default: master
+        description: 'Name of the branch you want to build from. We usually build from master'
+        name: GITHUB_BRANCH
+
+    builders:
+    - shell: !include-raw: scripts/common/get-node.sh
+#    - shell: jobs/scripts/common/bootstrap.sh $WORKSPACE/jobs/scripts/glusterd2/run-test.sh "ghprbPullId=${ghprbPullId}"
+    - shell: !include-raw: scripts/glusterfs/regression.sh
```
Comment (deepshikhaaa) [inline jobs/glusterfs-regression.yml]: We don't run a script directly on the slave machine. In order to have the script on the duffy node (provided by get-node.sh) we need to run bootstrap.sh which will scp the regression.sh on the given duffy node.

File: jobs/scripts/glusterfs/regression.sh (hunk 1 above)
Comment (amarts) [reply jobs/scripts/glusterfs/regression.sh]: Done

File: jobs/glusterfs-regression.yml (hunk 2 above)
Comment (amarts) [reply jobs/glusterfs-regression.yml]: Thanks, now resolved.

**Comment by deepshikhaaa** [comment]
Looks good to me. Merging it.

**Pull request merged**